src/WebUI/Common/KubeZeroData.tsx

```diff
@@ -29,35 +34,35 @@ export class KubeZeroData extends BaseComponent<IKubeZeroDataProps> {
     }
 
     public render(): JSX.Element {
+        const zeroDataElm: JSX.Element = (
+            <ZeroData
+                className={css("flex-grow", this.props.className)}
+                imageAltText={this.props.imageAltText || ""}
+                imagePath={this.props.imagePath || ""}
+                secondaryText={this._getTextArea()}
+            />
+        );
         return (
-            <Card titleProps={{ text: this.props.title }} className={css("flex-grow", "item-top-padding", "kube-list-content", this.props.className)}>
-                <ZeroData
-                    className="flex-grow"
-                    imageAltText={this.props.imageAltText || ""}
-                    imagePath={this.props.imagePath || ""}
-                    secondaryText={this._getTextArea()}
-                />
-            </Card>
+            this.props.renderOnCard ?
+                <Card titleProps={{ text: this.props.title }} className={css("flex-grow", "item-top-padding", "kube-list-content")}>
+                    {zeroDataElm}
+                </Card> :
+                zeroDataElm
         );
     }
 
-    public static _getDefaultZeroData(hyperLink: string, hyperLinkLabel: string, description: string,
-                                    additionalText: string, title?: string, className?: string): JSX.Element{
+    public static _getDefaultZeroData(zeroDataProps: IKubeZeroDataProps): JSX.Element{
         return (
             <KubeZeroData
-                imagePath={require("../zero_data.png")}
-                title={title}
-                hyperLink={hyperLink}
-                hyperLinkLabel={hyperLinkLabel}
-                descriptionText={description}
-                additionalHelpText={additionalText}
-                className={className}
+                imagePath={require("../../img/zero_data.png")}
+                {...zeroDataProps}
             />
         );
     }
 
     private _getTextArea(): JSX.Element {
         return (<div>
+            {this.props.primaryText ? <span className={this.props.primaryTextClassName? this.props.primaryTextClassName: "zerod-primary-text"}>{this.props.primaryText}</span>: null}<br />
```

Comment:
for readability can male following in another tmp variable

this.props.primaryTextClassName? this.props.primaryTextClassName: "zerod-primary-text"